### PR TITLE
US279 T287, 288 and 289 GenericForm support for all basic models

### PR DESCRIPTION
**Pull request by Tengelma**
### Overview
This is a pull request demonstrating the GenericForms component. It currently supports 9 out of 14 subforms. The remaining 5 will be tackled next sprint (or maybe later this Sprint if I have time) so this should be relatively stable for a little while.

The core changes:
- Introduces GenericForm.vue
- Modifies forms.js (the file containing form metadata) to include extra information related to each subform
- Reverts Vue Material to an older version to avoid our longstanding 'this.$el.validty' warnings on md-select changes
- Modifies the naming of some things to be more consistent with our data representation
- Abstracts model selection options to a new file model_choices.json

This PR does not include tests because I want to get some feedback on it before I dive into that.

### Includes:
See above

### Developer Checklist:
- [X] The code follows the Quality Policy file on Google Drive
- [ ] My code has been developer-tested and includes unit tests
- [X] I have considered proper use of exceptions
- [X] I have eliminated IDE warnings
- [X] I have included tasks associated with this pull request in the Sprint Progress Sheet

### Notes:
- This is being merged into the GenericForms branch to keep this breaking change out of dev
- A review could be you pulling it down and confirming that a) the subforms that have been implemented so far feel and behave like they used to, b) code quality is up to par, and c) you understand and are on board with this general pattern

### CODE REVIEW

File: client/src/constants/forms.js [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+//    Examples: Growth, HOD, Bias, Mass Definition, Halo Exclusion
+// Basic
+//    Description: Mostly numeric fields with no core parameters or special styling
+//    Examples: TracerProfile, HaloProfile, Tracer Concentration, Halo Concentration
+// Complex
+//    Description: Has core parameters with defined ranges
+//    Examples: HMF
+// Very Complex
+//    Description: Has core parameters with defined ranges, special styling, and/or special types
+//    Examples: Halo Model, Cosmology, Transfer,  Filter
```
Comment (aneuhold) [inline client/src/constants/forms.js:15]: This is helpful.

**Review by aneuhold — COMMENTED** [review]
This all looks good. Especially the renaming and the forms definitions. Is this supposed to take away forms at the moment though? It is hard to tell without the other forms if these errors are supposed to be coming up:

image

**Comment by Tengelma** [comment]
> This all looks good. Especially the renaming and the forms definitions. Is this supposed to take away forms at the moment though? It is hard to tell without the other forms if these errors are supposed to be coming up:
> 
> image

Yes, the forms that aren't supported by GenericForm have been excluded. The data is still there (so models will work) but you can't edit them right now.

I think the error you're seeing is related to having existing models with old naming conventions (initial state got changed a little bit so things are slightly different). Be sure to clear everything before you play with it.

**Review by aneuhold — APPROVED** [review]
Okay this all looks good then. Just cleared Redis DB and now it doesn't show those errors. Just the `DoubleField` error.

**Review by vperuzzi — APPROVED** [review]
This refactoring is a great improvement. All tests pass. No errors with Generic Form noted. Approving.

**Pull request merged**